test/src/main/java/org/apache/accumulo/test/functional/PerTableCryptoIT.java

```diff
@@ -0,0 +1,213 @@
+import org.apache.accumulo.core.spi.crypto.GenericCryptoServiceFactory;
+import org.apache.accumulo.core.spi.crypto.PerTableCryptoServiceFactory;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.accumulo.miniclusterImpl.MiniAccumuloConfigImpl;
+import org.apache.accumulo.test.TestIngest;
+import org.apache.accumulo.test.VerifyIngest;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PerTableCryptoIT extends AccumuloClusterHarness {
+  private final static Logger log = LoggerFactory.getLogger(PerTableCryptoIT.class);
+  private final String NO_ENCRYPT_STRING = "No on disk encryption detected";
+  private final String ENCRYPTED_STRING = "Encrypted with Params:";
+
+  @Override
+  public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    String keyPath =
+        System.getProperty("user.dir") + "/target/mini-tests/PerTableCrypto-testkeyfile";
+    cfg.setProperty(INSTANCE_CRYPTO_FACTORY, PerTableCryptoServiceFactory.class.getName());
+    cfg.setProperty(GENERAL_SERVICE_NAME_PROP, AESCryptoService.class.getName());
+    cfg.setProperty(AESCryptoService.KEY_URI_PROPERTY, keyPath);
+
+    // setup key file
+    try {
+      Path keyFile = new Path(keyPath);
+      FileSystem fs = FileSystem.getLocal(new Configuration());
+      fs.delete(keyFile, true);
+      if (fs.createNewFile(keyFile))
+        log.info("Created keyfile at {}", keyPath);
+      else
+        log.error("Failed to create key file at {}", keyPath);
+
+      try (FSDataOutputStream out = fs.create(keyFile)) {
+        out.writeUTF("sixteenbytekey"); // 14 + 2 from writeUTF
+      }
+    } catch (Exception e) {
+      log.error("Exception during configure", e);
+    }
+  }
+
+  @Test
+  public void testMultipleTablesAndWALs() throws Exception {
+    var tables = getUniqueNames(3);
+    Map<String,String> props = new HashMap<>();
+    NewTableConfiguration tableConfig = new NewTableConfiguration();
+    TableId tableId1;
+    TableId tableId2;
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      // maybe with more splits WAL will be easier to observe?
+      props.put(Property.TABLE_SPLIT_THRESHOLD.getKey(), "75K");
+
+      // table 0 plain text
+      c.tableOperations().create(tables[0]);
+      TableId tableId0 = TableId.of(c.tableOperations().tableIdMap().get(tables[0]));
+      VerifyIngest.VerifyParams params = new VerifyIngest.VerifyParams(getClientProps(), tables[0]);
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
+      // verify data was not encrypted
+      System.out.println("-------------" + checkTableEncryption(tableId0, false));
+
+      // encrypt table 1
+      props.clear();
```

Comment:
It looks like `props` is not used before it is cleared. Is this intended?

Reply:
That was holdover from some experimentation I was doing to try and detect encryption in the WALs. I think I can drop it now.